server/gokbg3/grails-app/controllers/wekb/WorkflowController.groovy

```diff
@@ -0,0 +1,287 @@
+          flash.error = "Invalid action type information: ${action_config.actionType}".toString()
+          break
+      }
+    }
+    else{
+      flash.error = "Unable to locate action config for ${params.selectedBulkAction}".toString()
+      log.warn("Unable to locate action config for ${params.selectedBulkAction}")
+      redirect(url: result.ref)
+    }
+  }
+
+  @Transactional
+  @Secured(['ROLE_EDITOR', 'IS_AUTHENTICATED_FULLY'])
+  def processTippRetire(){
+    log.debug("processTippRetire ${params}")
+    def retired_status = RefdataCategory.lookupOrCreate(RCConstants.KBCOMPONENT_STATUS, 'Retired')
+    def result = [:]
+
+    params.list('beforeTipps').each{ title_oid ->
+      log.debug("process ${title_oid}")
+      def tipp_obj = genericOIDService.resolveOID2(title_oid)
+      tipp_obj.status = retired_status
+      if (params.endDateSelect == 'select' && params.selectedDate){
+        tipp_obj.accessEndDate = params.date('selectedDate', 'yyyy-MM-dd')
+      }
+      else if (params.endDateSelect == 'now'){
+        tipp_obj.accessEndDate = new Date()
+      }
+      tipp_obj.save(flush: true, failOnError: true)
+    }
+    redirect(url: params.ref)
+  }
+
+  private def triggerSourceUpdateAllTitles(packages_to_update) {
+    triggerSourceUpdate(packages_to_update, true)
+  }
+
+  private def triggerSourceUpdate(packages_to_update, boolean allTitles=false) {
+    log.info("triggerSourceUpdate for Packages ${packages_to_update}..")
+    def user = springSecurityService.currentUser
+    def pars = [:]
+    def denied = false
+
+    boolean userAdmin = user.isAdmin()
+
+    if (packages_to_update.size() > 1){
+      flash.error = "Please select a single Package to update!"
+    }
+    else{
+      packages_to_update.each{ ptv ->
+        def pkgObj = Package.get(ptv.id)
+
+        if (pkgObj && pkgObj.source?.url){
+
+          if (accessService.checkEditableObject(pkgObj, userAdmin)) {
+            Set<Thread> threadSet = Thread.getAllStackTraces().keySet()
+            Thread[] threadArray = threadSet.toArray(new Thread[threadSet.size()])
+            boolean processRunning = false
+            threadArray.each { Thread thread ->
+              if (thread.name == 'triggerSourceUpdate' + pkgObj.id) {
+                processRunning = true
+              }
+            }
+
+            if(processRunning){
+              flash.error = 'A package update is already in progress. Please wait this has finished.'
+            }else {
+              executorService.execute({
+                Package aPackage = Package.get(pkgObj.id)
+                Thread.currentThread().setName('triggerSourceUpdate_' + pkgObj.id)
+                autoUpdatePackagesService.startAutoPackageUpdate(aPackage, !allTitles)
+              })
+
+              flash.success = "The package update for Package '${pkgObj.name}' was started. This runs in the background. When the update has gone through, you will see this on the Auto Update Info of the package tab."
```

Comment:
When the update has finished, it will be displayed at the "Auto Update Info" of the package tab.